// @tinymce/tinymce-angular/esm5/utils/Utils.js
/**
 * Copyright (c) 2017-present, Ephox, Inc.
 *
 * This source code is licensed under the Apache 2 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
import { validEvents } from '../editor/Events';
var bindHandlers = function (ctx, editor) {
    var allowedEvents = getValidEvents(ctx);
    allowedEvents.forEach(function (eventName) {
        var eventEmitter = ctx[eventName];
        editor.on(eventName.substring(2), function (event) { return ctx.ngZone.run(function () { return eventEmitter.emit({ event: event, editor: editor }); }); });
    });
};
var ɵ0 = bindHandlers;
var getValidEvents = function (ctx) {
    var ignoredEvents = parseStringProperty(ctx.ignoreEvents, []);
    var allowedEvents = parseStringProperty(ctx.allowedEvents, validEvents).filter(function (event) { return validEvents.includes(event) && !ignoredEvents.includes(event); });
    return allowedEvents;
};
var ɵ1 = getValidEvents;
var parseStringProperty = function (property, defaultValue) {
    if (typeof property === 'string') {
        return property.split(',').map(function (value) { return value.trim(); });
    }
    if (Array.isArray(property)) {
        return property;
    }
    return defaultValue;
};
var ɵ2 = parseStringProperty;
var unique = 0;
var uuid = function (prefix) {
    var date = new Date();
    var time = date.getTime();
    var random = Math.floor(Math.random() * 1000000000);
    unique++;
    return prefix + '_' + random + unique + String(time);
};
var ɵ3 = uuid;
var isTextarea = function (element) {
    return typeof element !== 'undefined' && element.tagName.toLowerCase() === 'textarea';
};
var ɵ4 = isTextarea;
var normalizePluginArray = function (plugins) {
    if (typeof plugins === 'undefined' || plugins === '') {
        return [];
    }
    return Array.isArray(plugins) ? plugins : plugins.split(' ');
};
var ɵ5 = normalizePluginArray;
var mergePlugins = function (initPlugins, inputPlugins) {
    return normalizePluginArray(initPlugins).concat(normalizePluginArray(inputPlugins));
};
var ɵ6 = mergePlugins;
// tslint:disable-next-line:no-empty
var noop = function () { };
var ɵ7 = noop;
var isNullOrUndefined = function (value) { return value === null || value === undefined; };
var ɵ8 = isNullOrUndefined;
export { bindHandlers, uuid, isTextarea, normalizePluginArray, mergePlugins, noop, isNullOrUndefined };
export { ɵ0, ɵ1, ɵ2, ɵ3, ɵ4, ɵ5, ɵ6, ɵ7, ɵ8 };
